### ISSUE TITLE
Update Menu Type do not clear the `page_id` data

### ISSUE THREAD

**Issue by kiddtang** [Laravel-Backpack/MenuCRUD#51]
# Bug report
Changing the type of the menu do not clear the previous field's content, the url generation is correct according to the type, but it causes unexpected result for relationship using "page_id"

Should consider clearing previous `type` data (i.e. `link` or `page_id`) after changing to the new `type`

### What I did:
Step 1: Create a Menu with type of `page_link`, Select a Page
Step 2: Edit the Menu and change the type to `internal_link` and Save

### What I expected to happen:
`page_id` data should be cleared to avoid unexpected result

### What happened:
![image](https://user-images.githubusercontent.com/26017547/67914163-43bdf980-fbca-11e9-8ccc-96b470a9bc6b.png)

**Comment by tabacitu** [comment]
I agree! Thank you for raising this issue @kiddtang . We'll get on this as soon as we launch Backpack 4.1 - until then that's the priority.

Cheers!
